dds/DCPS/transport/shmem/ShmemTransport.cpp

```diff
@@ -116,6 +115,7 @@ ShmemTransport::configure_i(TransportInst* config)
                     ACE_TEXT("no platform support for shared memory!\n")),
                    false);
 #else
+  ShmemInst* conf = dynamic_cast<ShmemInst*>(config);
```

Comment:
That's fine to change the title, and I don't think it's worth the effort to go through and separate this now, but please keep in mind for the future that it's much cleaner to handle these submissions as distinct PRs with a single purpose each.